server/services/transaction.service.js

```diff
@@ -284,26 +285,32 @@ export const getOneTransaction = async (transactionId, user) => {
 };
 
 export const addRemittance = async (remittanceInfo) => {
-  const transaction = await getTransactionById(remittanceInfo.transactionId).catch((error) => {
+  const transaction = await getOneTransaction(remittanceInfo.transactionId, {
```

Comment:
I think the `getOneTransaction` implementation is flawed. Since we are getting just one transaction, I don't think it makes sense for it to return an array 🤷🏽‍♂️ 

However, if fixing it is too much work, that it will be better to destructure the returned array so that you don't need to use [0]. This should be updated to

```suggestion
  const [transaction] = await getOneTransaction(remittanceInfo.transactionId, {
```

After updating, I think it will be safe to use `transaction` with `[0]`.